### PR TITLE
PCM-3818: Fixing Advanced Search init-related regressions

### PR DESCRIPTION
**Pull request by jtrantin**
@vrathee @gandhikeyur Please review.

https://projects.engineering.redhat.com/browse/PCM-3818

This was mostly caused by the session.js-related changes in securityService. Fix for that is already common module, while testing I found some other problems, so I fixed those too.

### CODE REVIEW

File: app/cases/controllers/advancedSearchController.js [hunk 1]
```diff
@@ -52,10 +52,18 @@ export default class AdvancedSearchController {
             });
 
             if ($scope.solrQuery != null && SOLRGrammarService.parseSafe($scope.solrQuery) != null) {
-                $timeout(() => {
+                const search = () => {
                     AdvancedCaseSearchService.clearColumns(); // reset columns just before search is started to prevent UI glitches
                     $scope.submitSearch(); // Delay the search, so that rha-filterselect is initiated
-                }, 1000);
```
Comment (jtrantin) [inline app/cases/controllers/advancedSearchController.js]: Now, we don't wait for some period of time until rha-filterselect does it's job, but instead filterselect fires an event when it initializes. This enables to chain these two initialization procedures in a much predictable way.

File: app/main.module.js [hunk 2]
```diff
@@ -77,6 +77,12 @@ const app = angular.module('RedhatAccess', [
     'RedhatAccess.escalation'
 ]);
 
+app.run(["$rootScope",function ($rootScope) {
+    Promise.setScheduler(function (cb) {
+        $rootScope.$evalAsync(cb);
+    });
+}]);
```
Comment (jtrantin) [inline app/main.module.js]: Note sure where this code disappeared (I believe we had it previously), but we need it when using bluebird with Angular.

**Comment by vrathee** [comment]
Reviewed!

**Comment by jtrantin** [comment]
Bumped common version.

**Pull request merged**